elide-core/src/main/java/com/yahoo/elide/core/RequestScopedTransaction.java

```diff
@@ -0,0 +1,14 @@
+/*
+ * Copyright 2016, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.core;
```

Comment:
I spent some time thinking about this.  So we have a situation today where the RequestScope takes the transaction as an argument to the constructor and the Transaction now needs access to the RequestScope.  This circular dependency doesn't seem right to me.  If you look at the RequestScope, the only reason it needs a transaction is to save dirty objects.  This save could easily (and more appropriately) been called on the transaction itself.  I think a bit of minor refactoring can fix this.  I would remove transaction from RequestScope, move 'save' from RequestScope to the transaction, and now transaction can take a RequestScope as an argument to the constructor.  Throughts?